host/client_integration_test.go

```diff
@@ -459,7 +459,7 @@ func (s *clientIntegrationSuite) TestClientDataConverter_WithChild() {
 
 	// to ensure custom data converter is used, this number might be different if client changed.
 	d := dc.(*testDataConverter)
-	s.Equal(3, d.NumOfCallToPayloads)
+	s.Equal(2, d.NumOfCallToPayloads)
```

Comment:
In older SDK versions the data converter was called 3 times but now it is 2. This is because logic was changed deep in the interceptor chains on when values are serialized and encoded and such. Previously we invoked the data converter on results even if there was an error (and most of the time they were nil). Now we don't do such a thing, resulting in fewer calls.